### ISSUE TITLE
Concatenation with empty tuple creates Union with 'nothing'

### ISSUE THREAD

**Issue by Cebtenzzre** [google/pytype#1090]
#### Software versions
Python 3.9.9
pytype 2021.12.15

#### Example
```python
i = 1
x, = (i,) + ()
reveal_type(x)  # Union[int, nothing]
reveal_type(x + 1)  # Union[Any, int]
```

#### Description
It seems like #549 was not completely fixed - the `nothing` type can still leak out. This has a snowball effect: Doing something with x tends to result in `Union[Any, <correct type>]`, which can make type narrowing less effective, so what should be `Dict[str, int]` can become `Union[Any, list]`, and then `--protocols` will assume the index type of `Union[slice, int]` instead of `str`, leading to confusing false positives.

**Comment by rchen152** [comment]
#549 was about "nothing" showing up in merge-pyi annotations; it's still a valid internal type, so I'm not surprised that it shows up in `reveal_type` output. But there is a bug here. `(1,) + ()` should still be `(1,)`; I think pytype is instead treating it as an addition of `Tuple[int, ...]` and `Tuple[nothing, ...]`.

**Comment by martindemello** [comment]
fixed by google/pytype@5fc73db
